core/node/metadata_calculator/src/helpers.rs

```diff
@@ -634,7 +634,10 @@ impl L1BatchWithLogs {
                     StorageKey::new(AccountTreeId::new(tree_write.address), tree_write.key);
                 TreeInstruction::write(storage_key, tree_write.leaf_index, tree_write.value)
             });
-            let reads = protective_reads.into_iter().map(TreeInstruction::Read);
+            let reads = protective_reads
+                .into_iter()
+                .sorted()
+                .map(TreeInstruction::Read);
```

Comment:
TBH, I think it would be safer to first collect protective reads and writes to a `Vec` and sort it then. It would require less non-local reasoning, and it looks like overhead would be reasonably small.

Reply:
Yeah, agree. Then comment won't be needed